pkg/workflows/sdk/compute.go

```diff
@@ -20,3 +26,40 @@ func (c *computeOutputCap[T]) Value() CapDefinition[T] {
 }
 
 var _ ComputeOutputCap[struct{}] = &computeOutputCap[struct{}]{}
+
+type ComputeConfig[C any] struct {
+	Config C
+}
+
+func (c *ComputeConfig[C]) ToMap() (map[string]any, error) {
+	wc, err := values.WrapMap(c.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	uc, err := wc.Unwrap()
+	if err != nil {
+		return nil, err
+	}
+
+	cm, ok := uc.(map[string]any)
+	if !ok {
+		return nil, errors.New("could not convert config into map")
+	}
```

Comment:
Do you think it's worth a quick check up front to see if C is already a map[string]any?

var cmp map[string, any]
var ok bool

if cm, ok = uc.(map[string]any); !ok {
   c.toMap() // this function is just the lines below.
}